test/oqs_test_evp_pkey_params.c

```diff
@@ -0,0 +1,613 @@
+
+/** \brief List of hybrid KEMs. */
+const char *kHybridKEMAlgorithms[] = {
+    "p256_frodo640aes",     "x25519_frodo640aes", "p256_frodo640shake",
+    "x25519_frodo640shake", "p384_frodo976aes",   "x448_frodo976aes",
+    "p384_frodo976shake",   "x448_frodo976shake", "p521_frodo1344aes",
+    "p521_frodo1344shake",  "p256_kyber512",      "x25519_kyber512",
+    "p384_kyber768",        "x448_kyber768",      "x25519_kyber768",
+    "p256_kyber768",        "p521_kyber1024",     "p256_mlkem512",
+    "x25519_mlkem512",      "p384_mlkem768",      "x448_mlkem768",
+    "x25519_mlkem768",      "p256_mlkem768",      "p521_mlkem1024",
+    "p384_mlkem1024",       "p256_bikel1",        "x25519_bikel1",
+    "p384_bikel3",          "x448_bikel3",        "p521_bikel5",
+    "p256_hqc128",          "x25519_hqc128",      "p384_hqc192",
+    "x448_hqc192",          "p521_hqc256",        NULL,
+};
+
+/** \brief Indicates if a string is in a given list of strings.
+ *
+ * \param list List of strings.
+ * \param s String to test.
+ *
+ * \return 1 if `s` is in `list`, else 0. */
+static int is_string_in_list(const char **list, const char *s)
+{
+    for (; *list != NULL && strcmp(*list, s) != 0; ++list)
+        ;
+    if (*list != NULL) {
+        return 1;
+    }
+    return 0;
+}
+
+/** \brief Indicates if a signature algorithm is hybrid or not.
+ *
+ * \param alg Algorithm name.
+ *
+ * \returns 1 if hybrid, else 0. */
+#define is_signature_algorithm_hybrid(_alg_) \
+    is_string_in_list(kHybridSignatureAlgorithms, (_alg_))
+
+/** \brief Indicates if an kem algorithm is hybrid or not.
+ *
+ * \param alg Algorithm name.
+ *
+ * \returns 1 if hybrid, else 0. */
+#define is_kem_algorithm_hybrid(_alg_) \
+    is_string_in_list(kHybridKEMAlgorithms, (_alg_))
+
+/** \brief A pair of keys. */
+struct KeyPair {
+    /** \brief The public key. */
+    uint8_t *pubkey;
+
+    /** \brief The public key length, in bytes. */
+    size_t pubkey_len;
+
+    /** \brief The private key. */
+    uint8_t *privkey;
+
+    /** \brief The private key length, in bytes. */
+    size_t privkey_len;
+
+    /** \brief Indicates if the pair of keys is from a quantum-resistant
+     * algorithm (1) or not (0). */
+    int is_pq;
+};
+
+/** \brief Frees the memory occupied by a KeyPair.
+ *
+ * \param kp Keypair to free. */
+static void keypair_free(struct KeyPair *kp)
+{
+    free(kp->pubkey);
+    free(kp->privkey);
+}
+
+/** \brief Initializes an OpenSSL top-level context.
+ *
+ * \returns The top-level context, or `NULL` if an error occurred. */
+static OSSL_LIB_CTX *init_openssl(void)
+{
+    OSSL_LIB_CTX *ctx;
+
+    if (!(ctx = OSSL_LIB_CTX_new())) {
+        fputs("failed to initialize a new `OSSL_LIB_CTX`\n", stderr);
+    }
+
+    return ctx;
+}
+
+/** \brief Loads the default provider.
+ *
+ * \param libctx Top-level OpenSSL context.
+ *
+ * \returns The default provider, or `NULL` if an error occurred. */
+static OSSL_PROVIDER *load_default_provider(OSSL_LIB_CTX *libctx)
```

Comment:
appears to be a general (test) utility function: Worthwhile adding to `test_common`?